enhancements/cluster-logging/logs-observability-openshift-io-apis.md

```diff
@@ -0,0 +1,602 @@
+  serviceAccount:
+    name:
+  collector:
+    resources:       #corev1.ResourceRequirements
+      limits:        #cpu, memory
+      requests:
+    nodeSelector:    #map[string]string
+    tolerations:     #corev1.Toleration
+  inputs:
+  - name:
+    type:                      #enum: application,infrastructure,audit
+    application:
+      selector:                #labelselector
+      includes:
+      - namespace:
+        container:
+      excludes:
+      - namespace:
+        container:
+      tuning:
+        ratelimitPerContainer:  #rate limit applied to each container selected by this input
+          recordsPerSecond:    #int  (no multiplier, a each container only runs on one node at a time.)
+    infrastructure:
+      sources: []              #enum: node,container
+    audit:
+      sources: []              #enum: auditd,kubeAPI,openshiftAPI,ovn
+    receiver:
+      type:                    #enum: syslog,http
+      port:
+      http:
+        format:                #enum: kubeAPIAudit , format of incoming data
+      tls:
+        cacert:
+          key:                 #the key in the resource
+          configmap:
+            name:                # the name of resource
+          secret:
+            name:                # the name of resource
+        certificate:
+          key:                   #the key in the resource
+          configmap:
+            name:                # the name of resource
+          secret:
+            name:                # the name of resource
+        key:
+          key:                   #the key in the resource
+          secret:
+            name:                # the name of resource
+        keyPassphrase:
+          key:                   #the key in the resource
+          secret:
+            name:                # the name of resource
+  filters:
+  - name:
+    type:                      #enum: kubeapiaudit, detectmultiline, parse, labels, drop, prune
+    kubeAPIAudit:
+    parse:
+  pipelines:
+    - inputRefs: []
+      outputRefs: []
+      filterRefs: []
+  outputs:
+  - name:
+    type:                    #enum: azureMonitor,cloudwatch,elasticsearch,googleCloudLogging,http,kafka,loki,lokiStack,splunk,syslog
+    tls:
+      cacert:
+        key:                 #the key in the resource
+        configmap:
+          name:                # the name of resource
+        secret:
+          name:                # the name of resource
+      certificate:
+        key:                   #the key in the resource
+        configmap:
+          name:                # the name of resource
+        secret:
+          name:                # the name of resource
+      key:
+        key:                   #the key in the resource
+        secret:
+          name:                # the name of resource
+      key:
+        key:                   #the key in the resource
+      keyPassphrase:
+        key:                   #the key in the resource
+      insecureSkipVerify:      #bool
```

Comment:
This is common TLS setting which short circuits certificate verification.  I'm not certain it can be represented any other way though I agree with you regarding bools.

>  Any chance to drop it in the new API?
@alanconway you and I have had this discussion in the past and I believe we always land on the "development" scenario where admins want this capability.  Our position has to been to lean into security but this is a knob that is useful during adoption

Reply:
Are there other TLS verification options? Such as system roots or specific keys?

You could have it as a combined DU, this has the nice benefit of preventing folks from setting keys when they would otherwise be ignored, because you skip the verification anyway
```
tlsVerification:
  method: SystemRoots | KeyFromSecret | InsecureSkipVerification
  keyFromSecret: <- only when method KeyFromSecret
    secret:
      name:
    key: ...

```